pkg/chartverifier/profiles/profile.go

```diff
@@ -88,30 +89,30 @@ func New(config *viper.Viper) *Profile {
 		}
 	}
 
-	var profile *Profile
 	vendorProfiles := profileMap[profileVendorType]
-	defaultProfile := vendorProfiles[0]
-
-	if len(vendorProfiles) > 1 {
-		profile = nil
-		for _, vendorProfile := range vendorProfiles {
-			if len(profileVersion) > 0 {
-				if semver.Compare(semver.MajorMinor(vendorProfile.Version), semver.MajorMinor(profileVersion)) == 0 {
-					profile = vendorProfile
-					break
+	if len(vendorProfiles) > 0 {
+		profileInUse = vendorProfiles[0]
+
+		if len(vendorProfiles) > 1 {
+			profileInUse = nil
+			for _, vendorProfile := range vendorProfiles {
+				if len(profileVersion) > 0 {
+					if semver.Compare(semver.MajorMinor(vendorProfile.Version), semver.MajorMinor(profileVersion)) == 0 {
+						profileInUse = vendorProfile
+						break
+					}
+				}
+				if semver.Compare(semver.MajorMinor(vendorProfile.Version), semver.MajorMinor(profileInUse.Version)) > 1 {
+					profileInUse = vendorProfile
 				}
-			}
-			if semver.Compare(semver.MajorMinor(vendorProfile.Version), semver.MajorMinor(defaultProfile.Version)) > 1 {
-				defaultProfile = vendorProfile
 			}
 		}
 	}
-
-	if profile == nil {
-		profile = defaultProfile
+	if profileInUse == nil {
```

Comment:
I would initialize profileInUse to this first then do the logic of the mapping to override or not.

Reply:
Done